pyrealm/demography/t_model_functions.py

```diff
@@ -79,8 +84,11 @@ def calculate_heights(
 
     if validate:
         _validate_t_model_args(pft_args=[h_max, a_hd], size_args=[dbh])
+        _validate_demography_array_arguments(
+            trait_args={"h_max": h_max, "a_hd": a_hd}, size_args={"dbh": dbh}
+        )
 
-    return h_max * (1 - np.exp(-a_hd * dbh / h_max))
+    return _enforce_2D(h_max * (1 - np.exp(-a_hd * dbh / h_max)))
```

Comment:
@j-emberton and @MarionBWeinzierl

I just wanted to pause here and get a sanity check before rolling this out further. Basically, there are two axes here: traits of a cohort (columns) and some measure of size for which something is being calculated. All of the functions accept arrays with varying dimensions, and the calculations can give rise to output arrays with 0, 1 or 2 dimensions. The calculations all work, but then the response dimensionality is unpredictable. My plan here is to promote all outputs to (n_traits, n_sizes), so that there is consistent outputs.

Does that seem sane?

Reply:
Yes I think the utility of having a consistent shape so that all downstream calcs can be simplified to assume this shape is reasonable.